### ISSUE TITLE
Alert emails being sent to locked accounts

### ISSUE THREAD

**Issue by matthewandrews** [AtlasOfLivingAustralia/alerts#32]
I've recently locked a number of accounts due to 'user not found' bounces, but it appears that alerts emails are still being sent to these addresses.  The accounts are still activated but are locked.

**Comment by nickdos** [comment]
Dependent on https://github.com/AtlasOfLivingAustralia/userdetails/issues/41

**Comment by nickdos** [comment]
Another commit for this issue be5a69b.

Test and close if deemed fixed.

**Comment by matthewandrews** [comment]
Based on the (lack of) bounces from alerts emails sent to undeliverable addresses in the last few weeks, this looks like it's working.

**Comment by matthewandrews** [comment]
Just got a pile of bounces - looks like these are still being sent to locked accounts after all. Possibly only for specific taxon search alerts (as opposed to the general 'new records', 'blog', etc).

**Comment by nickdos** [comment]
Thanks @matthewandrews (not really ;-) Could you paste some example emails that bounced so I can dig a bit deeper? Thanks

**Comment by matthewandrews** [comment]
Sure - a sample of alert query emails sent yesterday to locked accounts:

New records for Species: Sus scrofa  (sent Thu, 27 Sep 2018 17:15:24 +1000 to userid 83)
New records for Species: Leipoa ocellata (sent Thu, 27 Sep 2018 17:15:29 +1000 to userid 83)
New records for Species: Limax maximus (sent Thu, 27 Sep 2018 17:15:31 +1000 to userid 4037)
New records (sent Thu, 27 Sep 2018 17:15:01 +1000 to userid 83)
Citizen science records with images (sent Thu, 27 Sep 2018 17:15:04 +1000 to userid 1442)
New images (sent Thu, 27 Sep 2018 17:15:02 +1000 to userid 83)

**Comment by nickdos** [comment]
Deployed again, lets see how it goes this time.

**Comment by nickdos** [comment]
Hi @matthewandrews any bounces for the latest blogs email? If not can you close this pls?

**Comment by matthewandrews** [comment]
Looks good - no bounces from locked accounts (there were plenty of bounces, but they appear to be all from unlocked accounts - I've now locked the accounts that had hard bounces).

**Comment by matthewandrews** [comment]
Hmm, a blog alert today has been sent to some locked accounts.  Example: [https://alerts.ala.org.au/admin/user/7473](https://alerts.ala.org.au/admin/user/7473) (locked in userdetails; blog alerts turned on in alerts; alert email for blog was sent today).

**Comment by nickdos** [comment]
Update. I fixed the logging as well as the broken `UpdateEmailsJob` cron job and deployed changes. Re-running the `Update user emails with CAS` showed 24 addresses were updated (as opposed to 0 when run yesterday). Looking at the code the call to `AuthService.getUserId()` appears to be cached indefinitely, which would explain the accounts not being updated. It should be under EhCache but the Grails 3 changes buggered that so only basic cache plugin is used. Fix is to clear the cache when running the `updateUserEmails()` but before calling `AuthService`.
